### PR TITLE
fix #306234: added teleportation to smooth scrolling, for dealing with repeats + improved comments

### PR DESCRIPTION
**Pull request by SKefalidis**
Resolves: https://musescore.org/en/node/306234

Added teleportation (with options to enable/disable in the preferences) to deal with repeats. I also improved the comments.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [n\a] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by anatoly-os** [comment]
@SKefalidis thank you for the fix! Could you please make a PR containing both this and #6192 on top of `3.x`, not `master`?

**Comment by SKefalidis** [comment]
@anatoly-os rebased this and created #6193

File: mscore/scoreview.cpp [hunk 1]
```diff
@@ -714,6 +714,7 @@ void ScoreView::moveControlCursor(const Fraction& tick)
         }
         _timeElapsed += addition;
     } else { // reposition the cursor when the score is not playing
+        std::cout << "here" << std::endl;
```
Comment (anatoly-os) [inline mscore/scoreview.cpp:717]: @SKefalidis Could you please delete this?

Comment (SKefalidis) [reply mscore/scoreview.cpp]: Done.

**Comment by SKefalidis** [comment]
If anybody can explain the test failures I would be grateful. They seem to think that I changed parts of the code that I did not change.

**Pull request merged**